compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -6460,9 +6460,13 @@ private List<BLangNamedArgsExpression> checkProvidedErrorDetails(BLangErrorConst
             BType target = checkErrCtrTargetTypeAndSetSymbol(namedArgsExpression, expectedType);
 
             BLangNamedArgsExpression clone = nodeCloner.cloneNode(namedArgsExpression);
-            BType type = checkExpr(clone, target, data);
```

Comment:
The spec says, 
> The arg-name of every named-arg must be specified as the field-name of an individual-field-descriptor occurring in D, **unless there are no such field names**. 

Therefore, if we consider the scenarios with expression in named-arge having no user speficied CET,
```bal
type E1 error<record {}>;
type E2 error<map<string>>;
type E3 error<record {||}>;

error err1 = error("error message 1", fatal = false, msg = "msg"); // OK  [1]
error err2 = error E1("error message 2", fatal = false, msg = "msg"); // OK  [2]
error err3 = error E2("error message 3", fatal = "false", msg = "msg"); // OK [3]
error err3 = error E3("error message 3", fatal = "false", msg = "msg"); // ERROR [4]
```
from the `checkErrCtrTargetTypeAndSetSymbol` we get the the target type for named arg's expression,
[1] - `value:Clonable` from `map<value:Clonable>`
[2] - `anydata` from `record {}`
[3] - `string` from map<string>
[4] - `symbleTable.noType` from record {||} 

So, I think it is correct to type check with the target type. The whole logic here should be,
```bal
            BType target = checkErrCtrTargetTypeAndSetSymbol(namedArgsExpression, expectedType);
            checkExpr(namedArgsExpression, target, data);
            namedArgs.add(namedArgsExpression);
```

Reply:
Currently, we allow this because of the existing logic i guess. 
`error err = error("Error", code1 = [12, "A", 10.5, 1], code2 = {a: "A", b: [1, 2]});`

But, this should result in complication error,
>Each named-arg specifies a field of the error detail mapping; **the static type of each named-arg must be a subtype of value:Cloneable**.

Reply:
Resolving as per the offline discussion with @pcnfernando @prakanth97.